lms/djangoapps/teams/static/teams/js/spec/views/team_profile_header_actions_spec.js

```diff
@@ -0,0 +1,254 @@
+                );
+
+                // team has no space and current user is a not member of
+                // current team so we should see the correct message
+                expect(view.$('.action.action-primary').length).toEqual(0);
+                expect(view.$('.join-team-message').text().trim()).toBe(view.teamFullMessage);
+
+                // there should be no request made
+                expect(requests.length).toBe(0);
+            });
+
+            it('shows correct error message if user fails to join team', function () {
+                var requests = AjaxHelpers.requests(this);
+
+                // verify user_message
+                verifyErrorMessage(
+                    requests,
+                    JSON.stringify({'user_message': "Can't be made member"}),
+                    "Can't be made member",
+                    true
+                );
+
+                // verify generic error message
+                verifyErrorMessage(
+                    requests,
+                    '',
+                    'An error occurred. Try again.',
+                    true
+                );
+
+                // verify error message when json parsing succeeded but error message format is incorrect
+                verifyErrorMessage(
+                    requests,
+                    JSON.stringify({'blah': "Can't be made member"}),
+                    'An error occurred. Try again.',
+                    true
+                );
+            });
+
+            it('shows correct error message if initializing the view fails', function () {
+                // Rendering the view sometimes require fetching user's memberships. This may fail.
+                var requests = AjaxHelpers.requests(this);
+
+                // verify user_message
+                verifyErrorMessage(
+                    requests,
+                    JSON.stringify({'user_message': "Can't return user memberships"}),
+                    "Can't return user memberships",
+                    false
+                );
+
+                // verify generic error message
+                verifyErrorMessage(
+                    requests,
+                    '',
+                    'An error occurred. Try again.',
+                    false
+                );
+            });
+        });
+
+        describe('EditButton', function() {
+            var teamModelData,
+                view,
+                createAndAssertView;
+
+            createAndAssertView = function(showEditButton) {
+                teamModelData = createTeamModelData('aveA', 'avengers', createMembershipData('ma'));
+                view = createHeaderActionsView(1, 'ma', teamModelData, showEditButton);
+                expect(view.$('.action-edit-team').length).toEqual(showEditButton ? 1 : 0);
+            };
+
+            it('can render itself for staff or privileged user', function () {
```

Comment:
How about "renders for staff or privileged user" and "does not render for non-staff and non-privileged user"?